crates/rpc-types-eth/src/state.rs

```diff
@@ -30,6 +31,64 @@ pub struct AccountOverride {
     pub state_diff: Option<HashMap<B256, B256>>,
 }
 
+/// Helper type that bundles various overrides for EVM Execution.
+///
+/// By `Default`, no overrides are included.
+#[derive(Debug, Clone, Default)]
+pub struct EvmOverrides {
+    /// Applies overrides to the state before execution.
+    pub state: Option<StateOverride>,
+    /// Applies overrides to the block before execution.
+    ///
+    /// This is a `Box` because less common and only available in debug trace endpoints.
```

Comment:
if this type's only purpose is to be serialized, it could instead have borrows of each component?

Reply:
these are only really useful from the server POV and are intended to bundle overrides that are separate arguments

Reply:
so does this get deserialized as a unit? or does it get deser separately and then constructed?

Reply:
separately then constructed, so in the method implementation we can pass `overrides: EvmOverrides` instead of two Option arguments

Reply:
this smells like the API problem is minimizing diff for some existing-but-messy code. don't love it, but i won't block you merging it

Reply:
the issue here is that some endpoints support both override args, some only state, some only block, but a lot require the same setup, like eth_call, trace_call, call_many, estimate. 
this helper type makes it possible to reuse the same setup